apax/nodes/selection.py

```diff
@@ -179,12 +182,35 @@ def select_atoms(self, atoms_lst: typing.List[ase.Atoms]) -> typing.List[int]:
     def _get_selection_plot(
         self, atoms_lst: typing.List[ase.Atoms], indices: typing.List[int]
     ):
-        energies = np.array([atoms.calc.results["energy"] for atoms in atoms_lst])
-
-        if "energy_uncertainty" in atoms_lst[0].calc.results.keys():
-            uncertainty = np.array(
-                [atoms.calc.results["energy_uncertainty"] for atoms in atoms_lst]
+        has_calc = any(atoms.calc is not None for atoms in atoms_lst)
+        if not has_calc:
+            energies = np.zeros(len(atoms_lst))
+            Emean = 0.0
+        else:
+            raw_energies = np.array(
+                [atoms.calc.results["energy"] for atoms in atoms_lst if atoms.calc]
             )
+            Emean = np.mean(raw_energies)
+            energies = np.full(len(atoms_lst), Emean)
+            for i in range(len(atoms_lst)):
+                if atoms_lst[i].calc:
+                    energies[i] = atoms_lst[i].calc.results["energy"]
```

Comment:
`energies = np.array([atoms.get_potential_energy() if atoms.calc is not None else mean_energy for atoms in atoms_lst])`